File: Java Solutions/EraseSecurely.java
import java.util.Scanner;

public class EraseSecurely {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int num = sc.nextInt();
		sc.nextLine();
		StringBuilder begin = new StringBuilder(sc.nextLine());
		StringBuilder end = new StringBuilder(sc.nextLine());

		for (int j = 0; j < num; j++) {
			for (int i = 0; i < begin.length(); i++) {
				if (begin.charAt(i) == '1') {
					begin.setCharAt(i, '0');
				} else {
					begin.setCharAt(i, '1');
				}
			}
		}

		if (begin.toString().equals(end.toString())) {
			System.out.println("Deletion succeeded");
		} else {
			System.out.println("Deletion failed");
		}
	}
}

File: Python Solutions/EraseSecurely.py
a = int(input())
b = input()
c = input()
d = ""

for i in range(a):
    d = ""
    for k in range(len(c)):
        if b[k] == '1':
            d += b[k].replace('1', '0')
        else:
            d += b[k].replace('0', '1')
    b = d

if d == c:
    print('Deletion succeeded')
else:
    print('Deletion failed')
